### ISSUE TITLE
There is no option to wait until all Pods are ready

### ISSUE THREAD

**Issue by slaskawi** [arquillian/arquillian-cube#832]
Arquillian CUBE allows to wait for the service. However in some cases, it would be very useful to wait until all Pods (or specified Pods) become available and ready.

**Comment by lordofthejars** [comment]
In these cases you can use assertj extension https://fabric8.io/guide/testing.html

**Comment by slaskawi** [comment]
Yes, that's exactly what I'm implementing at the moment. 

My main concern is that you can wait for services specified in `arquillian.xml` ([here's an example](https://github.com/jboss-container-images/jboss-dataservices-image/blob/dataservices-dev/functional-tests/src/test/resources/arquillian.xml#L15)) and there is no such option for Pods.

**Comment by lordofthejars** [comment]
I see in case of OpenShift since we have the live checkness, but in kubernetes this concept does not exists. And you prefer to not use assertj approach but setting in `arquillian.xml`. Well any PR is more then welcome, are you planning to send a PR?

**Comment by slaskawi** [comment]
No, unfortunately I have some high priority tasks on my shoulders and probably won't have time to look into it in the nearest future. Please feel free to pick it up if you wish...

**Comment by Ladicek** [comment]
For anyone who stumbles upon this and is confused: this is for pods created by an external processes -- Cube already awaits readiness of all things it creates.

It should be fairly easy to implement, just extend the configuration and then adjust https://github.com/arquillian/arquillian-cube/blob/master/kubernetes/kubernetes/src/main/java/org/arquillian/cube/kubernetes/impl/SessionManager.java#L172

**Comment by spolti** [comment]
IIRC there is something similar in the ce-arq: https://github.com/jboss-openshift/ce-arq/blob/79ad6a9319d4fcaf3f85bde17e581a8982c97781/utils/src/main/java/org/jboss/arquillian/ce/utils/Containers.java

https://github.com/jboss-openshift/ce-arq/blob/8c598c8880316c031a6319b6e60ab62affc6930b/utils/src/main/java/org/jboss/arquillian/ce/adapter/AbstractOpenShiftAdapter.java#L178
